commcare_export/commcare_hq_client.py

```diff
@@ -232,6 +233,13 @@ def checkpoint(self, checkpoint_manager, paginator, batch, is_final):
                 logger.warning(
                     'Failed to get a checkpoint date from a batch of data.'
                 )
+        if isinstance(paginator, UCRPaginator):
+            cursor = paginator.next_page_params_from_batch(batch)['cursor'][0]
+            checkpoint_manager.set_checkpoint(
+                datetime.utcnow(),
+                is_final,
+                cursor=cursor,
+            )
```

Comment:
What do you think about pushing this to the paginator class?:

```
def set_checkpoint(checkpoint_manager, batch, is_final):
```

Reply:
@snopoke For that matter the whole `checkpoint` method can be pushed to the respective paginator classes (`DatePaginator` or `UCRPaginator`)

Reply:
@snopoke Something like [this](https://github.com/dimagi/commcare-export/pull/223/commits/90ba91789191a71ecb92bd1871933b41c94c01f9)?